packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -463,15 +461,11 @@ function preinit(href: string, options: PreinitOptions) {
           if (preloadProps) {
             adoptPreloadPropsForScript(scriptProps, preloadProps);
           }
-          instance = createElement(
-            'script',
-            scriptProps,
-            resourceRoot,
-            HTML_NAMESPACE,
-          );
-          markNodeAsResource(instance);
+          const ownerDocument = getDocumentFromRoot(resourceRoot);
+          instance = createHTMLElement('script', scriptProps, ownerDocument);
```

Comment:
This is not an inline script so it won't hit the special case in `createHTMLElement` so you can just use `ownerDocument.createElement`.